ndc-test/src/test_cases/fixture.rs

```diff
@@ -0,0 +1,467 @@
+                                && (!gen_config.exclude_fields.is_empty()
+                                    && gen_config.exclude_fields.contains(&key)))
+                        {
+                            continue;
+                        }
+
+                        let (property, property_value) = make_nested_field_recursive(
+                            gen_config,
+                            schema_response,
+                            rng,
+                            Box::new(field.r#type),
+                            Some(schema_type.clone()),
+                            &MakeNestedFieldState {
+                                current_depth: state.current_depth + 1,
+                                is_argument: state.is_argument,
+                            },
+                        )?;
+                        object_fields.insert(
+                            key.clone(),
+                            Field::Column {
+                                column: key.clone(),
+                                fields: property,
+                                arguments: make_query_arguments(
+                                    gen_config,
+                                    rng,
+                                    schema_response,
+                                    &field.arguments,
+                                )?,
+                            },
+                        );
+                        object_value.insert(key.clone(), property_value);
+                    }
+                    Ok((
+                        Some(NestedField::Object(NestedObject {
+                            fields: object_fields,
+                        })),
+                        Value::Object(object_value),
+                    ))
+                }
+            }
+        }
+        Type::Predicate { object_type_name } => {
+            let collection_type = schema_response
+                .object_types
+                .get(object_type_name.as_str())
+                .unwrap();
+
+            let (_, values) =
+                make_collection_fields(gen_config, schema_response, rng, collection_type)?;
+            let expression =
+                make_predicate_fixture(schema_response, rng, collection_type, vec![values])?;
+            Ok((None, json!(expression)))
+        }
+    }
+}
+
+pub fn make_predicate_fixture(
+    schema_response: &SchemaResponse,
+    rng: &mut SmallRng,
+    collection_type: &ndc_models::ObjectType,
+    rows: Vec<IndexMap<String, ndc_models::RowFieldValue>>,
+) -> Result<Option<Expression>> {
+    if let Some(context) = super::query::context::make_context(collection_type, rows)? {
+        let pred = predicate::make_predicate(
+            &configuration::TestGenerationConfiguration::default(),
+            schema_response,
+            &context,
+            rng,
+        )?;
+        Ok(pred.map(|p| p.expr))
+    } else {
+        Ok(None)
+    }
+}
+
+#[allow(deprecated)]
+fn make_scalar_value(
```

Comment:
Here is the problem with this approach for me, and why I didn't want to solve this here: there's no guarantee that we're generating semantically-meaningful values here. Just because a representation is string doesn't mean every string is meaningful, and just because a type is an array of strings doesn't mean every array of meaningful strings is meaningful.

I was planning to deal with this in a different way, by offering a way to generate test cases with missing values, but I'm not convinced it's worth it (compared to just hand-writing cases) given the amount of overhead for the user. 

For now I think this would be better as a separate tool or as part of the Go SDK's CLI, but if we continue it here, I'd like to hash out the approach in an RFC first.

Reply:
That makes sense. I'm also wondering if we should generate random values of just static ones. 
However, the purpose of this command is for fixture generation. The connector author also needs to modify expected snapshot values. It's just convenient for nested field generation. Connector authors take hours or days to copy-paste hundreds of snapshots.   

It will be great if you have some ideas to improve this generator.